src/jsp/jsp_sr.c

```diff
@@ -348,38 +404,68 @@ delay_load_dll_exception_filter (PEXCEPTION_POINTERS pep)
  */
 
 static void *
-jsp_get_create_java_vm_function_ptr (void)
+jsp_get_create_java_vm_function_ptr ()
 {
-  char *java_home = NULL;
-  char jvm_library_path[PATH_MAX] = { 0 };
-  void *libVM_p;
+  void *libVM_p = NULL;
 
-  libVM_p = dlopen (JVM_LIB_FILE, RTLD_NOW | RTLD_LOCAL);
-  if (libVM_p == NULL)
+  char *jvm_path = getenv ("JVM_PATH");
+  if (jvm_path != NULL)
     {
-      er_set (ER_WARNING_SEVERITY, ARG_FILE_LINE, ER_SP_JVM_LIB_NOT_FOUND, 1, dlerror ());
+      libVM_p = dlopen (jvm_path, RTLD_NOW | RTLD_LOCAL);
+      if (libVM_p != NULL)
+	{
+	  return dlsym (libVM_p, "JNI_CreateJavaVM");
+	}
+      else
+	{
+	  err_msgs.append ("\n\tFailed to load libjvm from 'JVM_PATH' environment variable: ");
+	  err_msgs.append ("\n\t\t");
+	  err_msgs.append (dlerror ());
+	}
+    }
+  else
+    {
+      err_msgs.append ("\n\tFailed to get 'JVM_PATH' environment variable");
+    }
 
-      java_home = getenv ("JAVA_HOME");
-      if (java_home != NULL)
+  char *java_home = getenv ("JAVA_HOME");
+  if (java_home != NULL)
+    {
+      char jvm_library_path[PATH_MAX];
+      err_msgs.append ("\n\tFailed to load libjvm from 'JAVA_HOME' environment variable: ");
```

Comment:
is it right that error message append to err_msgs, if jvm_home is not null ?

Reply:
The same reason with above